Browse malware info in the console (#3645)
chainguard-dev/edu: content/chainguard/libraries/browse.md

toc: true
---

> Chainguard Libraries provides access to millions of Java, JavaScript, and

Reviewer: can we use our new messaging here?

Chainguard Libraries is a malware-free catalog of JavaScript, Python, and Java dependencies that have passed through multiple layers of security controls, including malware and [greyware](https://www.chainguard.dev/unchained/the-expanding-threat-landscape-chainguard-now-scans-source-code-for-traditional-malware-and-greyware) scanning, building from source, cooldowns, and additional policies. Through the Chainguard Console, you can browse what's available, inspect package details, and evaluate dependencies before pulling them into your environment.

---

> Chainguard Libraries includes thousands of libraries and many more individual

Reviewer: "thousands" i think undersells it quite a bit. dont we have millions of artifacts?

Reviewer 2: Yes we have millions.

Author: This was old copy! I'll work on an update to this and audit other library docs pages that haven't been updated in a while

The list includes the following columns:

* **Name**: The full library name, excluding any version identifiers.
    * Python library names are simple strings, such as `setuptools` or
  `Flask-Admin`.
    * Java library names are the concatenation of the Maven
  coordinate values `groupId` and `artifactId`, separated by `:`. Examples are
  `org.springframework:spring-core` or `org.eclipse.jetty:jetty-http`.
* **Latest version**: The latest released and available version of the library
  and the total number of available versions.
* **Updated**: The most recent date when any version of this library was built

>   and published by Chainguard.

Reviewer 3: ```suggestion
* **Updated**: The most recent date when any version of this library was built
and published by Chainguard, or cached from the upstream fallback (when upstream fallback is enabled).
```

At the bottom of the page, see a total count of available libraries.

#### Upstream fallback details

At the top of the JavaScript ecosystem page, see whether [upstream fallback](/chainguard/libraries/overview/#upstream-fallback-and-controls) is enabled. For JavaScript, you can enable and disable upstream fallback in the Console or via `chainctl` commands.

> Upstream fallback and policy controls can be enabled for the Python and Java ecosystems via `chainctl` commands, but cannot currently be enabled via the Chainguard Console. Learn more in the [Libraries Overview documentation](/chainguard/libraries/overview/#upstream-fallback-and-controls). are available for Chainguard Libraries and can be enabled via `chainctl` commands. For JavaScript, in addition to the ability to enable upstream fallback via `chainctl`, you can also enable it in the Chainguard Console.

Reviewer 3: ```suggestion
The upstream fallback is available for JavaScript, Python, and Java Libraries. The upstream fallback can be enabled or disabled using the `chainctl libraries entitlements` commands, per ecosystem. Currently, the Chainguard Console can be used to enable, disable, and view the status of the upstream fallback for JavaScript only. At the top of the JavaScript page in the Console, you can see whether [upstream fallback](/chainguard/libraries/overview/#upstream-fallback-and-controls) is enabled.

Learn more in the [Libraries Overview documentation](/chainguard/libraries/overview/#upstream-fallback-and-controls). 
```

### View remediated libraries

[CVE remediation](/chainguard/libraries/cve-remediation/) is available for a

> subset of Chainguard Libraries for Python and Java. You can view remediated libraries in

Reviewer: can we put (beta) after Java. has caused some confusion with post-sales

Learn more about browsing remediations in [CVE remediation for Chainguard
Libraries](/chainguard/libraries/cve-remediation/#about-cve-remediation).

### View malware information

For the JavaScript and Python ecosystems, click the **Malware** tab to learn how many packages have been blocked by Chainguard due to malware or greyware detection.

> The list at the bottom of the page displays all blocked packages, including the package name, blocked version, the date it was blocked on, its MAL ID if available, and signals detected.

Reviewer 3: ```suggestion
The list at the bottom of the page displays all blocked packages from malware and greyware scanning. The list is in chronological order based on the time each package was blocked, starting with the most recently blocked. The columns displayed include the package name, blocked version, the date and time it was blocked on, its MAL ID if available, and signals detected through scanning.
```

Learn more about browsing remediations in [CVE remediation for Chainguard
Libraries](/chainguard/libraries/cve-remediation/#about-cve-remediation).

### View malware information

For the JavaScript and Python ecosystems, click the **Malware** tab to learn how many packages have been blocked by Chainguard due to malware or greyware detection.

The list at the bottom of the page displays all blocked packages, including the package name, blocked version, the date it was blocked on, its MAL ID if available, and signals detected.

> #### Malicious signal categories

Reviewer 3: @egibs could you review these public docs for defining the "signals detected" for malware/greyware? cc: @rossgordon-cg

Reviewer 2: We discussed consolidating this into maybe 4-5 categories. Are you both comfortable with this @egibs @angela-zhang ? Please confirm so sally can make the updates!

Reviewer 3: In just the docs or also in the API? We'd need to make some changes in the API layer if we want to narrow down the "Signals detected". I'm fine with narrowing down but we'd need to decide on which categories to keep or consolidate.

> #### Malicious signal categories

Reviewer 3: ```suggestion
#### Malware and greyware signal categories
```

> Some categories describe confirmed malicious behavior, while others describe greyware or supply-chain risk indicators that may justify blocking even when there is no public malware advisory. Categories include:

Reviewer: Next to each blocked version, Chainguard provides the reasons it was blocked when there is no public malware advisory.

> 

Reviewer: Want to confirm that the list below actually matches the list we will show in console. i thought we were using high order categories so there were fewer total.

Learn more about browsing remediations in [CVE remediation for Chainguard
Libraries](/chainguard/libraries/cve-remediation/#about-cve-remediation).

### View malware information

For the JavaScript and Python ecosystems, click the **Malware** tab to learn how many packages have been blocked by Chainguard due to [malware or greyware detection](/chainguard/libraries/overview/#malware-and-greyware-detection).

> The list at the bottom of the page displays automatically blocked packages, including the package name, blocked version, the date it was blocked on, its MAL ID if available, and signals detected.

Reviewer 3: ```suggestion
The list at the bottom of the page displays automatically blocked packages, including the package name, blocked version(s), the date it was blocked on, its MAL ID if available, and signals detected.
```

#### Malware and greyware signals detected

> Some signals describe confirmed malicious behavior, while others describe greyware or supply-chain risk indicators that may justify blocking even when there is no public malware advisory. Next to each blocked version, Chainguard provides the reason it was blocked when there is no public malware advisory. The detected signals fall under the following categories:

Reviewer 3: ```suggestion
Some signals describe confirmed malicious behavior, while others describe greyware or supply-chain risk indicators that may justify blocking even when there is no public malware advisory. Next to each blocked package, Chainguard provides the reason it was blocked when there is no public malware advisory. The detected signals fall under the following categories:
```

Reviewer 3: it could be all versions, or one version
